public/app/features/variables/custom/CustomVariableEditor.tsx

```diff
@@ -44,26 +41,23 @@ class CustomVariableEditorUnconnected extends PureComponent<Props> {
 
   render() {
     return (
-      <>
-        <VariableLegend>Custom options</VariableLegend>
-
-        <VariableTextAreaField
-          name="Values separated by comma"
-          value={this.props.variable.query}
-          placeholder="1, 10, mykey : myvalue, myvalue, escaped\,value"
-          onChange={this.onChange}
-          onBlur={this.onBlur}
-          required
-          width={52}
-          testId={selectors.pages.Dashboard.Settings.Variables.Edit.CustomVariable.customValueInput}
-        />
-        <VariableLegend>Selection options</VariableLegend>
-        <SelectionOptionsEditor
-          variable={this.props.variable}
-          onPropChange={this.onSelectionOptionsChange}
-          onMultiChanged={this.props.changeVariableMultiValue}
-        />
-      </>
+      <CustomVariableForm
+        query={this.props.variable.query}
+        multi={this.props.variable.multi}
+        allValue={this.props.variable.allValue}
+        includeAll={this.props.variable.includeAll}
+        onQueryChange={this.onChange}
+        onMultiChange={(event) =>
+          this.onSelectionOptionsChange({ propName: 'multi', propValue: event.currentTarget.checked })
+        }
+        onIncludeAllChange={(event) =>
+          this.onSelectionOptionsChange({ propName: 'includeAll', propValue: event.currentTarget.checked })
+        }
+        onAllValueChange={(event) =>
+          this.onSelectionOptionsChange({ propName: 'allValue', propValue: event.currentTarget.value })
+        }
+        onQueryBlur={this.onBlur}
```

Comment:
why is this not just `<CustomVariableEditor variable={this.props.variable} />`

Reply:
ah, is component shared between old and new architecture?

Reply:
Yes, this is shared, this is the point where we break between old and new to make it agnostic to data structure